polygon/p2p/downloader.go

```diff
@@ -0,0 +1,129 @@
+package p2p
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/ledgerwatch/log/v3"
+	"golang.org/x/sync/errgroup"
+
+	"github.com/ledgerwatch/erigon-lib/gointerfaces/sentry"
+	"github.com/ledgerwatch/erigon/core/types"
+	"github.com/ledgerwatch/erigon/eth/protocols/eth"
+	"github.com/ledgerwatch/erigon/rlp"
+)
+
+const reqRespTimeout = 5 * time.Second
+
+var invalidDownloadHeadersRangeErr = errors.New("invalid download headers range")
+
+type RequestIdGenerator func() uint64
+
+type Downloader interface {
```

Comment:
I dont want to include the word Header in this because I plan to use this same component for getting block bodies too in the future. Having a separate fetcher/downloader per data model feels too fine grained to me. I can change the naming to fetcher

Reply:
addressed - https://github.com/ledgerwatch/erigon/pull/9470